### PR TITLE
Do not warn when the name key is used in the lr_scheduler dict

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes bug reported in slack cc @pvnieo
```
I followed the example provided here: https://pytorch-lightning.readthedocs.io/en/latest/generated/pytorch_lightning.callbacks.LearningRateMonitor.html?highlight=LearningRateMonitor

def configure_optimizer(self):
    optimizer = torch.optim.Adam(...)
    lr_scheduler = {'scheduler': torch.optim.lr_scheduler.LambdaLR(optimizer, ...)
                    'name': 'my_logging_name'}
    return [optimizer], [lr_scheduler]

but I'm getting the following warning:

python3.8/site-packages/pytorch_lightning/utilities/distributed.py:45: RuntimeWarning: Found unsupported keys in the lr scheduler dict: ['name']
```

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [n/a] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._

### CODE REVIEW

File: pytorch_lightning/callbacks/lr_monitor.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def _find_names(self, lr_schedulers) -> List[str]:
         names = []
         for scheduler in lr_schedulers:
             sch = scheduler['scheduler']
-            if 'name' in scheduler:
+            if scheduler.get('name') is not None:
```
Comment (Borda) [inline pytorch_lightning/callbacks/lr_monitor.py:160]: this shall be the same, right?

File: pytorch_lightning/trainer/optimizers.py [hunk 2]
```diff
@@ -94,6 +94,7 @@ def configure_schedulers(self, schedulers: list, monitor: Optional[str] = None):
         lr_schedulers = []
         default_config = {
             'scheduler': None,
+            'name': None,  # no custom name
```
Comment (Borda) [inline pytorch_lightning/trainer/optimizers.py:97]: is this needed as with `my_dictionary.get('missing-key')` returns None by default

**Review by Borda — COMMENTED** [review]
can you clarify what is the fix? :]

File: pytorch_lightning/trainer/optimizers.py (hunk 2 above)
Comment (carmocca) [reply pytorch_lightning/trainer/optimizers.py:97]: Yes because we use this `default_config` dict to validate the user input just below.

File: pytorch_lightning/callbacks/lr_monitor.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/callbacks/lr_monitor.py:160]: No because `'name' in {'name: None} == True` and we want to avoid that.

**Comment by carmocca** [comment]
> can you clarify what is the fix? :]

These are the lines which produce the warning:

https://github.com/PyTorchLightning/pytorch-lightning/blob/2c3d43dcb503cf2f8fc8a4b30d4ab6f986f348f1/pytorch_lightning/trainer/optimizers.py#L106-L108

so to avoid it, we need to include `name` in the `default_config` dict which is used for user input validation and for the parsing of the scheduler structure.

File: pytorch_lightning/callbacks/lr_monitor.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/callbacks/lr_monitor.py:160]: yes, but below you adding it to the dictionary...

Comment (carmocca) [reply pytorch_lightning/callbacks/lr_monitor.py:160]: `scheduler['name']` would work, but not `is in`

**Comment by rohitgr7** [comment]
what's this `name` used for BTW?

**Comment by carmocca** [comment]
> what's this `name` used for BTW?

To have custom names in the `LearningRateMonitor`. So `my_custom_name` instead of `lr-Adam` (default if using Adam)

**Comment by rohitgr7** [comment]
add this to docs too? https://pytorch-lightning.readthedocs.io/en/latest/lightning_module.html#configure-optimizers

File: pytorch_lightning/core/lightning.py [hunk 3]
```diff
@@ -1001,21 +1001,22 @@ def configure_optimizers(
...
                     'frequency': 1, # The frequency of the scheduler
                     'reduce_on_plateau': False, # For ReduceLROnPlateau scheduler
                     'monitor': 'val_loss', # Metric for ReduceLROnPlateau to monitor
-                    'strict': True # Whether to crash the training if `monitor` is not found
+                    'strict': True, # Whether to crash the training if `monitor` is not found
+                    'name': None, # Custom name for LearningRateMonitor to use
                 }
 
-            If user only provides LR schedulers, then their configuration will set to default as shown above.
+            Only the `scheduler` key is required, the rest will be set to the defaults above.
```
Comment (rohitgr7) [inline pytorch_lightning/core/lightning.py:1019]:
```suggestion
            Only the ``scheduler`` key is required, the rest will be set to the defaults above.
```

**Review by rohitgr7 — APPROVED** [review]
LGTM

**Review by SkafteNicki — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
